### PR TITLE
test(source): refactor upsert avro tests

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- move to `source_inline` and use risedev env vars to be easier to run locally
- put setup script in slt
- merge related stuff into one slt, and make it more readable

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by xxchan** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/17044?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#17044** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/17044?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#17032** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/17032?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @xxchan and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: e2e_test/source_inline/kafka/avro/name_strategy.slt [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+# TODO: Uncomment this when we add test data kafka key with format `"ID":id`
+# statement ok
+# CREATE TABLE upsert_avro_json (
+#   PRIMARY KEY("ID")
+# )
+# WITH (
+# connector = 'kafka',
+#  properties.bootstrap.server = '${RISEDEV_KAFKA_BOOTSTRAP_SERVERS}',
+# topic = 'upsert_avro_json')
+# FORMAT UPSERT ENCODE AVRO (schema.registry = '${RISEDEV_SCHEMA_REGISTRY_URL}');
```
Comment (xxchan) [inline e2e_test/source_inline/kafka/avro/name_strategy.slt:48]: I don't actually understand what's the commented test mean. Just kept it

Comment (tabVersion) [reply e2e_test/source_inline/kafka/avro/name_strategy.slt:48]: @yuhao-su  https://github.com/risingwavelabs/risingwave/pull/13023 Do you still remember why we commented out the test case?

**Review by tabVersion — APPROVED** [review]

File: e2e_test/source_inline/kafka/avro/name_strategy.slt (hunk 1 above)
Comment (yuhao-su) [reply e2e_test/source_inline/kafka/avro/name_strategy.slt:48]: We used to parse kafka key with Json format. That's way it required the key looks like `"ID": id`. But now we parse kafka key as `bytes`, so I think we can uncomment this now.

Comment (tabVersion) [reply e2e_test/source_inline/kafka/avro/name_strategy.slt:48]: try enable in https://github.com/risingwavelabs/risingwave/pull/17062

**Pull request merged**